src/library/dao/cuedao.cpp

```diff
@@ -208,10 +208,8 @@ void CueDAO::saveTrackCues(
     QStringList cueIds;
     cueIds.reserve(cueList.size());
     for (const auto& pCue : cueList) {
-        // New cues (without an id) must always be marked as dirty
-        DEBUG_ASSERT(pCue->getId().isValid() || pCue->isDirty());
         // Update or save cue
-        if (pCue->isDirty()) {
+        if (!pCue->getId().isValid() || pCue->isDirty()) {
```

Comment:
The removed debug assertion was failing very rarely (1 or 2 times) but I was not able to reproduce it nor have any idea what could have caused it.

@Holzhaus Any insights?

Reply:
I was able to trigger it reliably, that was the exactly why I made this PR ^^ but as said, I can also revert the last commit for the alternative solution.